linetools/isgm/utils.py

```diff
@@ -239,6 +241,73 @@ def xhtbl_from_components(components, ztbl=None, NHI_obj=None):
     tbl = Table()
     #
 
+def complist_from_table(table):
+    """
+    Returns a list of AbsComponents from an input table.
+
+    Parameters
+    ----------
+    table : Table or QTable
+        Table with component information (each row must correspond
+        to a component)
+
+    Returns
+    -------
+    complist : list
+        List of AbsComponents defined from the input table.
+
+    Notes
+    -----
+    Mandatory column names: 'RA_deg', 'DEC_deg', 'ion_name', 'z_comp', 'vmin_kms', 'vmax_kms'
```

Comment:
I have tended to not include units in the column names, i.e.
  RA instead of RA_deg
  vmin instead of vmin_kms

I'm not opposed to your approach, though.  This is more of a comment.